lib/services/local/generic/generic.go

```diff
@@ -121,6 +121,30 @@ func (s *Service[T]) WithPrefix(parts ...string) *Service[T] {
 	}
 }
 
+// CountResources will return a count of all resources in the prefix range.
+func (s *Service[T]) CountResources(ctx context.Context) (int, error) {
+	rangeStart := backend.ExactKey(s.backendPrefix)
+	rangeEnd := backend.RangeEnd(rangeStart)
+
+	count := 0
+	for {
+		result, err := s.backend.GetRange(ctx, rangeStart, rangeEnd, int(s.pageLimit))
+		if err != nil {
+			return 0, trace.Wrap(err)
+		}
+
+		numItems := len(result.Items)
+		if numItems == 0 {
+			break
+		}
+
+		count += numItems
+		rangeStart = backend.NextKey(result.Items[numItems-1].Key)
+	}
```

Comment:
Replace this loop with `backend.IterateRange`, this loop is subtly broken: `backend.NextKey` is not actually the next possible key in the backend, just the next possible key that doesn't have the same prefix, so if a page ends at `foo` this will continue with a GetRange starting from `fop`, skipping over `foobar`.

Reply:
Thank you for catching this. I always seem to be running into errors like this.

Reply:
The `GetRange` interface is far too easy to misuse like this, unfortunately. 😢